futures-core/src/lib.rs

```diff
@@ -73,7 +73,7 @@ macro_rules! pin_mut {
         // Shadow the original binding so that it can't be directly accessed
         // ever again.
         #[allow(unused_mut)]
-        let mut $x = unsafe { ::std::mem::PinMut::new_unchecked(&mut $x) };
+        let mut $x = unsafe { ::core::mem::PinMut::new_unchecked(&mut $x) };
```

Comment:
Could you implement it via a reexport of libcore and see if the strategy @cramertj proposes [here](https://github.com/rust-lang-nursery/futures-rs/pull/1051#discussion_r198942744) works?

Reply:
Yes, it seems to work.